src/Integration.UnitTests/Telemetry/TelemetryManagerTests.cs

```diff
@@ -452,5 +452,157 @@ private static void CheckExpectedLanguages(TelemetryData data, params string[] e
+
+            var installationTime = DateTimeOffset.Parse("2020-05-31T09:30:00.000+00:00");
+            var mockTimeProvider = new Mock<ICurrentTimeProvider>();
+            mockTimeProvider.Setup(x => x.LocalTimeZone).Returns(TimeZoneInfo.Utc);
+            resetMocks(installationTime);
+
+            // 1. Create -> initial values set
+            var testSubject = CreateManager(mockTimeProvider.Object);
+        
+            telemetryData.NumberOfDaysOfUse.Should().Be(0);
+            telemetryRepositoryMock.Verify(x => x.Save(), Times.Once);
+
+            // 2. First update on day 1 -> days of use updated and saved
+            var firstUpdateTime = installationTime.AddMinutes(1);
+            resetMocks(firstUpdateTime);
+
+            testSubject.Update();
+
+            telemetryData.NumberOfDaysOfUse.Should().Be(1);
+            telemetryRepositoryMock.Verify(x => x.Save(), Times.Once);
+
+            // 3. Second update on day 1 -> no change to days of use
+            var secondUpdateTime = installationTime.AddMinutes(2);
+            resetMocks(secondUpdateTime);
+
+            testSubject.Update();
+
+            telemetryData.NumberOfDaysOfUse.Should().Be(1);
+            telemetryRepositoryMock.Verify(x => x.Save(), Times.Never);
+
+            // 4. First update on day 2 -> days of use updated and saved
+            var secondDay_firstUpdateTime = installationTime.AddDays(1);
+            resetMocks(secondDay_firstUpdateTime);
+
+            testSubject.Update();
+
+            telemetryData.NumberOfDaysOfUse.Should().Be(2);
+            telemetryRepositoryMock.Verify(x => x.Save(), Times.Once);
+
+            // 5. Second update on day 2 -> no change to days of use
+            var secondDay_secondUpdateTime = secondDay_firstUpdateTime.AddMinutes(1);
+            resetMocks(secondDay_secondUpdateTime);
+
+            testSubject.Update();
+
+            telemetryData.NumberOfDaysOfUse.Should().Be(2);
+            telemetryRepositoryMock.Verify(x => x.Save(), Times.Never);
+
+            void resetMocks(DateTimeOffset newNow)
+            {
+                mockTimeProvider.Setup(x => x.Now).Returns(newNow);
+                telemetryRepositoryMock.Reset();
+                telemetryRepositoryMock.Setup(x => x.Data).Returns(telemetryData);
+            }
+        }
+
+        [TestMethod]
+        [DataRow("2020-05-31T20:30:00.000+00:00", "2020-06-01T06:30:00.000+08:00", 8, false)] // regression test for #1434
+        [DataRow("2020-06-01T04:30:00.000+00:00", "2020-06-01T06:30:00.000+00:00", 0, false)] // same day -> don't update
+        [DataRow("2020-05-31T20:30:00.000+00:00", "2020-06-01T06:30:00.000+00:00", 0, true)] // different day -> update
+        [DataRow("2020-06-01T04:30:00.000+00:00", "2020-08-23T02:30:00.000+08:00", 8, true)] // many days later -> update
+        public void Update_IfNotSameDayAsLastUpdate_DaysOfUseIsIncremented(string lastSavedDate, string now, int utcOffset, bool expectedIncrement)
```

Comment:
As I mentioned in a different PR, I think this test really highlights the fact that "utcOffset" parameter is odd. The timezone should be taken from the `now` parameter, as it won't be possible for the `now` to be +08:00, but the utcOffset to be different than 8.
I think the whole calculation code could've been simplified without the additional time zone parameter. 
I wouldn't fix it now, but If we touch the code again I think it's worth considering to simplify it.